app/controllers/admins/orders_controller.rb

```diff
@@ -2,7 +2,6 @@ module Admins
   class OrdersController < BaseController
```

Comment:
why are you calling `authorize!` method in the `deliver` action? In my understanding you have this already covered in the `BaseController`

Reply:
I thought that I need to additionally authorize admin actions against users submitting forms via changing urls or other attacks. I was wrong. Now I see that validation in_before actions with csrf protects against this. 
I have removed this methods.